pytest/lib/mocknet.py

```diff
@@ -847,39 +844,52 @@ def reset_data(node, retries=0):
             )
 
 
-def start_genesis_updater_script(script, genesis_filename_in,
-                                 genesis_filename_out, chain_id,
-                                 validator_nodes, rpc_nodes, done_filename,
-                                 epoch_length, node_pks, increasing_stakes,
-                                 num_seats):
+def start_genesis_updater_script(
+        script, genesis_filename_in, genesis_filename_out, records_filename_in,
+        records_filename_out, config_filename_in, config_filename_out, chain_id,
+        validator_nodes, rpc_nodes, done_filename, epoch_length, node_pks,
+        increasing_stakes, num_seats, sharding, all_node_pks, node_ips):
     return '''
         cd {dir}
         rm -f {done_filename}
-        nohup ./venv/bin/python {script} {genesis_filename_in} {genesis_filename_out} {chain_id} {validator_nodes} {rpc_nodes} {done_filename} {epoch_length} {node_pks} {increasing_stakes} {num_seats} 1> genesis_updater.out 2> genesis_updater.err < /dev/null &
+        nohup ./venv/bin/python {script} {genesis_filename_in} {genesis_filename_out} {records_filename_in} {records_filename_out} {config_filename_in} {config_filename_out} {chain_id} {validator_nodes} {rpc_nodes} {done_filename} {epoch_length} {node_pks} {increasing_stakes} {num_seats} {sharding} {all_node_pks} {node_ips} 1> genesis_updater.out 2> genesis_updater.err < /dev/null &
```

Comment:
There’s so many parameters at this point that I would wonder if it’d
be better to do a join:

    cmd = ' '.join(shlex.quote(str(arg)) for arg in (
        'nohup',
        './venv/bin/python',
        script,
        genesis_filename_in,
        genesis_filename_out,
        records_filename_in,
        records_filename_out,d
        config_filename_in,
        config_filename_out,
        chain_id,
        ','.join(validator_nodes),
        ','.join(rpc_nodes),
        done_filename,
        epoch_length,
        ','.join(node_pks),
        increasing_stakes,
        num_seats,
        sharding,
        ','.join(all_node_pks),
        ','.join(node_ips)
    ))
    return f'''
        cd {shlex.quote(dir)}
        rm -f -- {shlex.quote(done_filename)}
        {cmd} >genesis_updater.out 2>genesis_updater.err </dev/null &
    '''